File: ClientApp/src/app/model/homeVm.ts
import { ProductVm } from "./productVm";
import { CategoryVm } from "./categoryVm";
import { CategoryAttributeVm } from "./categoryAttributeVm";
import { MegaMenuItem, MenuItem, TreeNode } from "primeng/api";
import WsMenuItem from "./wsMenuItem";

export class HomeVm {

    Products: ProductVm[];
    CurrentCategory: CategoryVm;
    Categories: CategoryVm[];
    CategoriesTreeNode: TreeNode[];
    CurrentAttributes: CategoryAttributeVm[];
    MenuItems: MenuItem[];
    MegaItems: MegaMenuItem[];
    CurrentCategoryBreadCrumb: WsMenuItem;
}

File: Model/ViewModel/HomeVM.cs
using System.Collections.Generic;
using targeteo.pl.Model.ViewModel;

namespace targeteo.pl.Model.ViewModel
{
    public class HomeVM
    {
        public CategoryVM CurrentCategory { get; set; }
        public ICollection<ProductMainPageVM> Products { get; set; }
        public ICollection<CategoryVM> Categories { get; set; }

        public ICollection<TreeNodeVm> CategoriesTreeNode { get; set; }
        public List<CategoryAttributeVM> CurrentAttributes { get; set; }

        public ICollection<MenuItem> MenuItems { get; set; }

        public MenuItem CurrentCategoryBreadCrumb { get; set; }
    }
}